tests/test_generate_cpp.cpp

```diff
@@ -0,0 +1,42 @@
+// Introducing google test because cannot get nala to build with bazel - too much effort for too little reward
+// to create interopability between python, c, make, and bazel
+#include <gtest/gtest.h>
+
+#include "motohawk.h"
+
+// Copy struct pack/unpack in test_basic.motohawk_example_message_t
+TEST(motohawk, struct_unpack) {
+    uint8_t buffer[] = {"\xc0\x06\xe0\x00\x00\x00\x00\x00"};
+    ExampleMessage m(&buffer[0], 8u);
+    EXPECT_EQ(1, m.Enable_raw());
+    EXPECT_EQ(32, m.AverageRadius_raw());
+    EXPECT_EQ(55, m.Temperature_raw());
+}
+
+TEST(motohawk, struct_pack) {
+    uint8_t buffer[8];
+    ExampleMessage m(&buffer[0], 8u);
+    // TODO if this buffer isn't cleared/initialized then the test will fail
```

Comment:
Thoughts on this?

Reply:
Maybe I'm missing something, but shouldn't we validate the output buffer in this particular test case after we construct and pack the message? Since it tests the packing functionality (reverse of struct_unpack). For the initialization, seems like the initializing/zeroing the buffer in the Message class seems a bit safer but I don't feel strongly about it...

Reply:
Yeah, I think you should compare the output to a known "serialized" buffer at the end.

Reply:
done